README.md

```diff
@@ -164,3 +166,31 @@ npm run benchmark-report
+
+## Hardware Requirements
+
+Running the Archive Node API requires a hardware configuration that can handle the load placed on the server by user requests, with the main load being on the database. The following hardware requirements are recommended for optimal performance:
+
+CPU: The server should have a multi-core CPU, preferably with at least four cores, to handle the concurrent user requests and database transactions. A higher core count CPU will help reduce response times and ensure faster processing of queries.
+
+Memory: Sufficient memory is required to handle the increased load on the database. It is recommended to have at least 4GB of RAM.
+
+Network: A high-speed internet connection is recommended to reduce latency and improve query performance. The network infrastructure should be capable of handling concurrent user requests and database transactions.
+
+### Increased Load on the Database
+
+The Archive Node API server will place a heavy load on the database due to the nature of the complexity of the Archive Node database schema. To fetch events and action data related to a zkApp, there are multiple complex JOINs involved to aggregate the requested data. This can cause an increased load on the database and slow down response times.
+
+To mitigate the load on the database, it is recommended to take advantage of the multi-host connection to utilize multiple Archive Node databases. By supplying additional read-only databases for the API to connect to, queries will be spread out evenly across all databases which will help with response time. Additionally, considerations to use more powerful hardware on existing Archive Node databases should be considered.
+
+To get an accurate sense of what hardware requirements are needed, you can utilize the [benchmarking suite](#benchmarking) to see how many requests can be handled. Simply deploy your Archive Node database to an accessible endpoint, start the server, specify the URL to use for the Archive Node database, and run the benchmarking suite to see how higher requests affect response time.
+
+## Why are multiple blocks shown when getting event/action data at highest block of the network?
+
+One issue with this design is deciding how to return data from the tip of the network. At the tip of the network, each block producer submits their block to be included in the Mina network. Because Mina has weaker finality, it is difficult to know which block should be returned when querying for event/action data. The OCaml Mina daemon uses a selection algorithm to estimate the chain strength of each submitted block and then decides which block to move forward with.
```

Comment:
> One issue ... weaker finality...

The reader focuses on negative from the start: "issue with this design", would avoid this.

Would suggest starting with background on why this may occurs, discussing the two possible solutions short term & long term (long term we can use the chain selection algorithm, but are awaiting two pieces of data to be added to the archive node). In the near term, we've implemented this solution....